### PR TITLE
select feature in map and datatable

### PR DESCRIPTION
**Pull request by michaelmattig**

### CODE REVIEW

**Comment by michaelmattig** [comment]
[Daten selektieren in Tabelle](https://trello.com/c/mdHYt0V3/50-daten-selektieren-in-tabelle)

File: projects/wave-core-new/src/lib/datatable/table/table.component.ts [hunk 1]
```diff
@@ -33,10 +33,14 @@ export class DataTableComponent implements OnInit, AfterViewInit, OnDestroy, OnC
...
     readonly layerTypes = ResultTypes.VECTOR_TYPES;
 
+    selectedFeature$ = new BehaviorSubject<FeatureSelection>({feature: undefined});
+
     dataSource = new FeatureDataSource();
     displayedColumns: Array<string> = [];
+    featureColumns: Array<string> = [];
 
     protected layerDataSubscription: Subscription = undefined;
+    protected selectedLayerSubscription: Subscription = undefined;
```
Comment (jdroenner) [inline projects/wave-core-new/src/lib/datatable/table/table.component.ts:43]: rename to selectedFeatureSubscription

File: projects/wave-core-new/src/lib/datatable/table/table.component.ts [hunk 2]
```diff
@@ -46,6 +50,25 @@ export class DataTableComponent implements OnInit, AfterViewInit, OnDestroy, OnC
...
             this.emptyTable();
         }
+
+        this.selectedLayerSubscription = this.projectService.getSelectedFeatureStream().subscribe((selection) => {
+            this.selectedFeature$.next(selection);
+
+            if (!!this.paginator) {
+                for (let i = 0; i < this.dataSource.data.length; i++) {
+                    const feature = this.dataSource.data[i];
+                    if (feature.ol_uid === selection.feature) {
```
Comment (jdroenner) [inline projects/wave-core-new/src/lib/datatable/table/table.component.ts:60]: is this a feature or a id?

File: projects/wave-core-new/src/lib/map/map-container/map-container.component.ts [hunk 3]
```diff
@@ -332,111 +336,70 @@ export class MapContainerComponent implements AfterViewInit, OnChanges, OnDestro
...
+        });
     }
 
-    private attachUserSelectToMap(): void {
-        if (!this.userSelect) {
-            // called too early
-            return;
+    private performSelection(selection: FeatureSelection): void {
+        if (!!selection.feature) {
+            // TODO: avoid going through all layers
```
Comment (jdroenner) [inline projects/wave-core-new/src/lib/map/map-container/map-container.component.ts:358]: please create a task for this

File: projects/wave-core-new/src/lib/map/map-container/map-container.component.ts [hunk 4]
```diff
@@ -332,111 +336,70 @@ export class MapContainerComponent implements AfterViewInit, OnChanges, OnDestro
...
-            map = this.maps[0];
-        } else {
-            const selectedLayerIndex = this.mapLayers.map((layer) => layer.mapLayer).indexOf(this.selectedOlLayer);
-            if (selectedLayerIndex >= 0) {
-                const inverseIndex = this.maps.length - selectedLayerIndex - 1;
-                map = this.maps[inverseIndex];
-            } else {
-                map = undefined; // actually, something went wrong
+    private resetSelection(): void {
+        // TODO: avoid going through all layers
```
Comment (jdroenner) [inline projects/wave-core-new/src/lib/map/map-container/map-container.component.ts:376]: please create a task for this

**Comment by jdroenner** [comment]
please check if you try to select a feature or a layer. Also, create tasks for todos

File: projects/wave-core-new/src/lib/map/map-container/map-container.component.ts (hunk 3 above)
Comment (michaelmattig) [reply projects/wave-core-new/src/lib/map/map-container/map-container.component.ts:358]: https://trello.com/c/kJ7YjkT2

File: projects/wave-core-new/src/lib/map/map-container/map-container.component.ts (hunk 4 above)
Comment (michaelmattig) [reply projects/wave-core-new/src/lib/map/map-container/map-container.component.ts:376]: https://trello.com/c/kJ7YjkT2

File: projects/wave-core-new/src/lib/datatable/table/table.component.ts (hunk 2 above)
Comment (michaelmattig) [reply projects/wave-core-new/src/lib/datatable/table/table.component.ts:60]: it's an id. I initially had `feature` and `layer` as attributes of the `selection`, both Ids, but the layer matching is currently done via brute force (cf https://trello.com/c/kJ7YjkT2). Should i rename it to `feature_id`?

File: projects/wave-core-new/src/lib/datatable/table/table.component.ts (hunk 1 above)
Comment (michaelmattig) [reply projects/wave-core-new/src/lib/datatable/table/table.component.ts]: a9118f18b483cec650eb659b904b16d3f73d8fe2

**Pull request merged**